ignite/handlers/param_scheduler.py

```diff
@@ -842,7 +905,7 @@ def print_lr():
 
     def __init__(
         self,
-        lr_scheduler: PyTorchLRScheduler,
+        lr_scheduler: Union[PyTorchLRScheduler, _CosineAnnealingWarmRestarts],
```

Comment:
This should be without `_CosineAnnealingWarmRestarts`
```suggestion
        lr_scheduler: PyTorchLRScheduler,
```

You can annotate `self.lr_scheduler: Union[PyTorchLRScheduler, _CosineAnnealingWarmRestarts] = lr_scheduler` and then handle `isinstance(lr_scheduler, torch.optim.lr_scheduler.CosineAnnealingWarmRestarts)` case.